__test__/pages/forgotpassword.spec.js

```diff
@@ -0,0 +1,169 @@
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import RequestPasswordResetPage, { validateEmail } from '@pages/forgotPassword';
+import { BrowserRouter, Switch } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import {
+  render,
+  fireEvent,
+  waitForDomChange,
+  waitForElement,
+  wait
+} from '@testing-library/react';
+
+const mockStore = configureMockStore([thunk]);
+let store;
+const initialSate = {
+  isPending: false,
+  isSuccess: false,
+  error: null
+};
+
+const props = {
+  isLoading: false,
+  showSuccess: false,
+  errorMessage: null,
+  handleSubmit: jest.fn()
+};
+
+const renderWithEnzymes = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return mount(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Switch>
+          <RequestPasswordResetPage {...props} />
+        </Switch>
+      </BrowserRouter>
+    </Provider>
+  );
+};
+
+const renderWithRTL = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return {
+    ...render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Switch>
+            <RequestPasswordResetPage {...props} />
+          </Switch>
+        </BrowserRouter>
+      </Provider>
+    ),
+    store
+  };
+};
+
+describe('Test forgot password page', () => {
+  it('should have a email address field', () => {
+    const wrapper = renderWithEnzymes();
+    const field = wrapper.find('input').first();
+    expect(field.exists()).toBe(true);
+    expect(field.props().name).toEqual('email');
+  });
+
+  it('should have a submit button', () => {
+    const wrapper = renderWithEnzymes();
+    const button = wrapper.find('button').first();
+    expect(button.exists()).toBe(true);
+  });
+
+  it('should render forgot password page', () => {
+    const wrapper = renderWithEnzymes();
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should update password field on change', () => {
+    const wrapper = renderWithEnzymes();
+    const passwordInput = wrapper.find("input[name='email']");
+    passwordInput.simulate('change', {
+      persist: () => {},
+      target: {
+        name: 'email',
+        value: 'tejumoladavid@gmail.com'
+      }
+    });
+    expect(passwordInput.html()).toMatch('tejumoladavid@gmail.com');
+  });
+
+  it('should submit a valid form', () => {
+    const { getByTestId } = renderWithRTL(initialSate);
+
+    const emailInputElement = getByTestId('email');
+    fireEvent.change(emailInputElement, {
+      target: { value: 'tejumoladavid@gmail.com' }
+    });
+    fireEvent.blur(emailInputElement);
+
+    const formNode = getByTestId('submit-form');
+    fireEvent.submit(formNode);
```

Comment:
There seems to be noway to assert if this form was actually submitted after firing the event. Perhaps a change in the colour of the button or if the action ignited by the submit button is actually called

Reply:
Cool. I will look into this. Thanks